src/main/java/com/petros/bibernate/session/SessionFactoryImpl.java

```diff
@@ -1,12 +1,15 @@
 package com.petros.bibernate.session;
 
+import com.petros.bibernate.datasource.BibernateDataSource;
 import com.petros.bibernate.exception.BibernateException;
-import org.h2.jdbcx.JdbcDataSource;
 
 import javax.sql.DataSource;
 
 public class SessionFactoryImpl implements SessionFactory {
 
+    public static final String DATABASE_URL = "jdbc:h2:mem:test;DB_CLOSE_DELAY=-1;";
+    public static final String DATABASE_USERNAME = "sa";
+    public static final String DATABASE_PASSWORD = "";
```

Comment:
Have you considered using the fourth parameter, DRIVER_CLASS_NAME? If you do, you can create a datasource of the same type that was passed in this parameter. For example, Hikari supports this functionality using reflection. 
![image](https://user-images.githubusercontent.com/43843525/226163726-88e4f9d9-9fbb-4c35-a6dd-940d9d310a44.png)

![image](https://user-images.githubusercontent.com/43843525/226163940-6322f1bd-4351-412c-9621-1f1cdfcad37a.png)

It's worth exploring as a potential optimization

Reply:
The current implementation does not require this. I think the fourth parameter can be added while working on the "support hikari pool" task

Reply:
I'm just trying to convince you to get rid of the incorrect initialization of the H2 datasource inside the SessionFactoryImpl, either by explicitly passing the datasource through the constructor or initializing it through reflection. However, you seem hesitant to make this change despite its benefits from all perspectives :smiley:

Reply:
But initialisation of the H2 is a temporal solution. Instead of these constants, we will pass properties from configuration (when Configuration task will be merged). And we won't be dependent on particular database, because can set any db url in property file.
<img width="736" alt="image" src="https://user-images.githubusercontent.com/68471610/226172704-39978572-c092-4943-9e1a-25206213d021.png">

Reply:
Yes, but the current temporary solution is not working. You won't be able to create the SessionFactory right now. That's why I'm trying to convince you to use a different temporary solution, such as passing the datasource through the constructor, which can also be removed later (although I would recommend keeping it).